src/server/nonClassEvent/nonClassEvent.service.ts

```diff
@@ -56,4 +68,16 @@ export class NonClassEventService {
       .orderBy('p.area', 'ASC');
     return nonClassEvents.getMany();
   }
+
+  public async createWithNonClassEvents(parent: Partial<NonClassParent>):
+  Promise<NonClassParent> {
+    const semesters = await this.semesterRepository.find({});
+
+    return this.parentRepository.save({
+      ...parent,
+      nonClassEvents: semesters.map((semester) => ({
+        ...this.eventRepository.create(semester),
```

Comment:
Ah yes - well spotted. 

Also, I would have expected that to raise some kind of error from typescript since `.create` expects `DeepPartial<NonClassEvent>` and `semester` is of type `Semester`.

Reply:
Fixed this in 3b726db4bde990c7ccf88d68b4db815f047202a3